sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -216,21 +209,20 @@ def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
         self._receive_start_time = self._receive_start_time or time.time()
         deadline = self._receive_start_time + (max_wait_time or 0)  # max_wait_time can be None
         if len(self._message_buffer) < max_batch_size:
+            # TODO: the retry here is a bit tricky as we are using low-level api from the amqp client.
+            #  besides, the retry will create a new AMQPClient + new link properties (offset)
+            #  so we need to create new amqp client or update the link properties of existing one
+            #  I'd like to add more optional steps into the retry policy of amqp lib
+            #  to allow updating existing properties for recreation of a client if link/connection error happens
```

Comment:
_Should_ we be creating a new client?
Maybe the same client should recreate its connection?
A new client certainly shouldn't be needed just for a new link

Reply:
I agree, reusing the same client is the preferred approach.

however, I have not yet come up with a design on how to deal with callback mode retry as we're only calling the `client.do_work` here which has limitation.

maybe I would add a new method into amqp client to include sophisticated logic for retry handling in callback mode..

Reply:
Is this needed for this preview?
If not - let's have a chat about what's needed here and how we can cleanly incorporate it.

Reply:
I'm afraid it is needed -- otherwise, the behavior is somewhat breaking because per retry would receive duplicate messages not from the last received event's offset when error happened.

Reply:
Okay - well we can do whatever is needed to make the preview happy - but ultimately I don't think this should exist as a separate method - we should make the default path successful.